sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/analysis/Analyzer.scala

```diff
@@ -1010,7 +1014,16 @@ class Analyzer(
                 s"(valid range is [1, ${aggs.size}])")
           case o => o
         }
-        Aggregate(newGroups, aggs, child)
+        // If an aggregateExpression is integer, after replaced with this aggregateExpression, the
+        // groupExpression will be considered as an ordinal still, so we should filter it.
+        var filterGroups: Seq[Expression] = Nil
+        newGroups.foreach { ng =>
+          ng match {
+            case a: Alias => if (!isIntLiteral (a.child)) filterGroups :+= ng
+            case _ => filterGroups :+= ng
+          }
+        }
+        Aggregate(filterGroups, aggs, child)
```

Comment:
Please see the comments at https://github.com/apache/spark/blob/75b168fd30bb9a52ae223b6f1df73da4b1316f2e/sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/optimizer/Optimizer.scala#L1254-L1257. We should not make the grouping expressions empty.